exec/java-exec/src/main/codegen/templates/DrillVersionInfo.java

```diff
@@ -0,0 +1,97 @@
+/*******************************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+<@pp.dropOutputFile />
+
+<@pp.changeOutputFile name="/org/apache/drill/common/util/DrillVersionInfo.java" />
+
+<#include "/@includes/license.ftl" />
+
+package org.apache.drill.common.util;
+
+import java.text.MessageFormat;
+import java.text.ParseException;
+
+/**
+ * This file is generated with Freemarker using the template src/main/codegen/templates/DrillVersionInfo.java
+ */
+public class DrillVersionInfo {
+  private static final String VERSION = "${maven.version}";
+
+  private static final int MAJOR_VERSION;
+  private static final int MINOR_VERSION;
+  private static final int PATCH_VERSION;
+
+  static {
+      final int[] parseResults;
+      try {
+        parseResults = parseVersion(VERSION);
+        if (parseResults == null || parseResults.length < 3) {
+          throw new AssertionError("An error occured when parsing Drill Version");
+        }
+      } catch(ParseException e) {
+        throw new AssertionError("An error occured when parsing Drill Version", e);
```

Comment:
just curious, why not use defaults when we can't parse the version ? depending on when this exception is thrown, I'm not sure it will be easy for the user to figure out what went wrong